### ISSUE TITLE
_cleantarget_ generates too many commands

### ISSUE THREAD

**Issue by Smismi11** [codingo/Interlace#83]
I wanted to scan a list of CIDR IPs from a text file. I had trouble saving the output of each command to a text file. For example 

interlace -tL cidr.txt --no-cidr -c "amass enum -cidr _target_ >> _target_.txt"

does not work because of the "/" in the output file. While looking through the source code I found _cleantarget_ as variable. so I thought 

interlace -tL cidr.txt --no-cidr -c "amass enum -cidr _target_ >> _cleantarget_.txt"

has to work. Idk if _cleantarget_ is supported because it is not mentioned in the documentation.
But the problem here is that when I run the command with _cleantarget_ Interlace does not create x (x = number of lines in cidr.txt) commands but x*x commands (for example 64 instead of 8 commands).

I think the problem is that in the function _process_clean_targets every dirty_target gets replaced every time for every command, instead of replacing only the 1 dirty_target of that 1 command.
So the loop is executed to many times. It works without _cleantarget_ because you always check in the function add_task if the task is already in the list. But even for the else condition the "add_task(command, tasks, temp)" function gets called as many times as dirty_target exists instead of only once.

**Comment by codingo** [comment]
@prodigysml has been having a look into this already (tag for comments). I believe he may already have a fix in mind.

**Comment by Regala** [comment]
Hi! Wondering if there is any update on this? Thanks 🙏

**Comment by prodigysml** [comment]
Hey! Sorry about the wait. I'll take a look at this again asap.

**Comment by Regala** [comment]
Thank you so much!

**Comment by prodigysml** [comment]
Hey @Regala, is there any chance you have some test data I can use to test with?

**Comment by Regala** [comment]
@prodigysml for example:
interlace -tL hosts.txt -threads 2 -c "echo 'doing task for _target_ and saving output to _cleantarget_'" --no-bar 

where hosts.txt:
a.acme.com
b.acme.com

results in:
```
[13:28:07] [THREAD] [echo 'doing task for http://a.acme.com and saving output to a.acme.com'] Added to Queue
[13:28:07] [THREAD] [echo 'doing task for http://a.acme.com and saving output to b.acme.com'] Added to Queue
[13:28:07] [THREAD] [echo 'doing task for http://b.acme.com and saving output to a.acme.com'] Added to Queue
[13:28:07] [THREAD] [echo 'doing task for http://b.acme.com and saving output to b.acme.com'] Added to Queue
Generated 4 commands in total
doing task for http://a.acme.com and saving output to a.acme.com
doing task for http://b.acme.com and saving output to a.acme.com
doing task for http://a.acme.com and saving output to b.acme.com
doing task for http://b.acme.com and saving output to b.acme.com
```

Where it should have generated 2 commands only instead of 4.

Thanks!

**Comment by prodigysml** [comment]
Awesome! Thanks for that test data @Regala. I'll keep you posted.

**Comment by prodigysml** [comment]
Ok, so quick update. The bug that I see there is, Interlace is trying to create different combinations of the target. This is definitely a logic bug. Working on a fix.

**Comment by prodigysml** [comment]
Thank you very much for using our software and raising this issue @Regala and @Smismi11. Apologies for the delay on this. Issue should now be patched in version 1.7.5.
